docs/nodemcu-lrm.md

```diff
@@ -0,0 +1,331 @@
+
+#### lua_getstate
+
+`  lua_State * lua_getstate();`         [-0, +0, -]
+
+returns the main thread `lua_State` record.  Used in CBs to initialise `L` for subsequent API call use.
+
+#### lua_pushrotable
+`  void lua_pushrotable (lua_State *L, ROTable *p);`         [-0, +1, -]
+
+Pushes a ROTable onto the stack.
+
+#### lua_pushstringsarray
+`  int lua_pushstringsarray (lua_State *L, int opt);`         [-0, +1, m]
+
+Pushes an array onto the stack containing all strings in the specified strings table. If `opt` = 0 then the RAM table is used, else if `opt` = 1 and LFS is loaded then the LFS  table is used, else `nil` is pushed onto the stack.
+
+Returns a status boolean 1 = table pushed. 
+
+#### lua_stripdebug
+`  int lua_stripdebug (lua_State *L, int level);`         [-1, +0, e]
+
+This function has two modes.  A value is popped off the stack.  
+-  If this value is `nil`, then the default strip level is set to this if `level` in the range 0 to 2, otherwise the current default level is returned.
+-  If this is a Lua function (in RAM rather than in LFS), then the prototype hierarchy within the function is stripped of debug information to the specified level.
+
+### lua_writestring
+
+`  void lua_writestring(const char *s, size_t l); /* macro */`
+
+Writes a string `s` of length `l` to `stdout`.  Note that any output redirection will be applied to the string.
+ 
+### lua_writestringerror
+
+`  void lua_writestringerror(const char *s, void *p).  /* macro */`
+
+Writes an error with Cstring format specifier `s` and parameter `p` to `stderr`.  Note on the ESP devices this error will always be sent to `UART0`; output redirection will not be applied.
+
+### The Debug Interface
+
+#### lua_getstrings
+
+`  int lua_getstrings (lua_State *L, int opt);`         [-0, +1, m]
+
+If the ROM string table is selected and no LFS is present then `nil` is pushed onto the stack.  Otherwise the specified string table is scanned to build a list of strings in it. The resulting table is pushed onto the stack.
+-  `opt` evaluated as a boolean; if `true` then the ROM string table is listed, else the RAM string table
+-  returns the number of strings in returned table.
+
+### Auxiliary Library Functions and Types 
+
+Note that the LRM defines an auxiliary library which contains a set of functions that assist in coding convenience and economy.  These are strictly built on top of the basic API, and are defined in `lauxlib.h`.  By convention all auxiliary functions have the prefix `luaL_` so module code should only contain `lua_XXXX()` and `luaL_XXXX()` data declarations and functions.  And since the `lauxlib.h` itself incudes `lua.h`, all C modules should only need to `#include "lauxlib.h"` in their include preambles.
+
+NodeMCU adds some extra auxiliary functions above those defined in the LRM.
+
+#### luaL_lfsreload
+
+`  int  lua_lfsreload (lua_State *L);`         [-1, +1, -]
```

Comment:
Didn't the return type get changed to `void` above?

Reply:
At the moment it restarts (and so never returns on a successful load, but it returns an error string (e.g. file does not exist) on error.  Personally I'd prefer it to be a `void`and to throw errors, but returning the error was the consensus when we had the discussion back when.

Reply:
Yes, but as part of this PR it does change to a `void`. See line 202 in `lflash.c` and 290 in `lauxlib.h`.